src/openvic-simulation/economy/FactoryProducer.cpp

```diff
@@ -37,3 +44,64 @@ fixed_point_t FactoryProducer::get_average_profitability_last_seven_days() const
 
 	return sum / (1 + profit_history_current);
 }
+
+void FactoryProducer::produce(
+	const fixed_point_t input_modifier, const fixed_point_t throughput_modifier, const fixed_point_t output_modifier
```

Comment:
I imagine the country calls some method on each state, which calls this method for each factory.
The country & state create the modifier values here.